### PR TITLE
feat: Use resumablejs to upload data larger than 2mb in chunks

### LINKED ISSUE #2121: Send large data to Layman as multipart
# Feature Request

**Is your feature request related to a problem? Please describe.**
All data are currently sent in one request regardless its size. That runs into the request size limits on the server.

**Describe the solution you'd like**
Adopt the asynchronous file upload for larger data (lets say > 2 MB, which is the default threshold for request size). Layman describes the process in its docs https://github.com/LayerManager/layman/blob/master/doc/async-file-upload.md

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

Use resumableJs library to upload data larger than 2mb to Layman

## Related issues or pull requests

closes #2121 

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [ ] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/src/assets/locales/cs.json [hunk 1]
```diff
@@ -48,7 +48,8 @@
       "errorWhileRequestingLayers": "Chyba při žádání vrstev ze serveru",
       "someErrorHappened": "Nastala nespecifikovaná chyba nebo je chybný vstup",
       "thereWasErrorWhile": "Nastala chyba při syntaktické analýze odpovědi Capabilities z dané URL",
-      "urlInvalid": "Chybějící nebo neplatná adresa URL požadované vrstvy!"
+      "urlInvalid": "Chybějící nebo neplatná adresa URL požadované vrstvy!",
+      "srsNotSupported": "Vybraný souřadnicový systém (CRS/SRS) není podporován. Pokračujte poskytnutím souboru ve formátu EPSG:3857 nebo EPSG:4326."
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/cs.json:52]:
```suggestion
      "srsNotSupported": "Vybraný souřadnicový systém (CRS/SRS) není podporován. Pokračujte poskytnutím souboru v systému EPSG:3857 nebo EPSG:4326."
```

File: projects/hslayers/src/assets/locales/sk.json [hunk 2]
```diff
@@ -48,7 +48,8 @@
       "errorWhileRequestingLayers": "Chyba pri žiadaní vrstiev zo servera",
       "someErrorHappened": "Nastala nešpecifikovaná chyba alebo je chybný vstup",
       "thereWasErrorWhile": "Nastala chyba pri syntaktickej analýze odpovede Capabilities z danej URL",
-      "urlInvalid": "Chýbajúca alebo neplatná URL požadovanej vrstvy!"
+      "urlInvalid": "Chýbajúca alebo neplatná URL požadovanej vrstvy!",
+      "srsNotSupported": "Vybraný súradnícový systém (CRS/SRS) nie je podporovaný.Ak chete pokračovať, zadajte súbor vo formáte EPSG:3857 alebo EPSG:4326."
```
Comment (jmacura) [inline projects/hslayers/src/assets/locales/sk.json:52]: dtto as in CZ (I don't want to fuck up the declension 😄 )

File: projects/hslayers/src/components/add-data/file/shp/add-data-file-layer.directive.html [hunk 3]
```diff
@@ -70,18 +70,22 @@
         <div class="hs-ows-shp-advanced" *ngIf="advancedPanelVisible">
             <div class="form-group">
                 <label class="capabilities_label control-label">{{'ADDLAYERS.srs' | translate}}</label>
-                <input class="form-control" name="srs" [(ngModel)]="srs" />
+                <input class="form-control" name="srs" [(ngModel)]="srs"/>
```
Comment (jmacura) [inline projects/hslayers/src/components/add-data/file/shp/add-data-file-layer.directive.html:73]: Ok, but why?

File: projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts [hunk 4]
```diff
@@ -72,15 +96,25 @@ export class HsAddDataFileShpService {
...
+            } else {
+              resolve(data);
+            }
           } else {
             reject(data);
           }
         })
         .catch((err) => {
-          this.hsLog.error(err);
+          console.error(err);
```
Comment (jmacura) [inline projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts:117]: hsLog is our wrapper around console object, why replacing it?

File: projects/hslayers/src/components/save-map/layman.service.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable prettier/prettier */
```
Comment (jmacura) [inline projects/hslayers/src/components/save-map/layman.service.ts:1]: what was wrong with prettier in this file? I prefer to disable single lines if the rules are going to be messy somehow.

File: projects/hslayers/src/components/add-data/file/shp/add-data-file-layer.directive.html (hunk 3 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/add-data/file/shp/add-data-file-layer.directive.html:73]: :D good question. Must have been unintentional.

File: projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts (hunk 4 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts:117]: Without hs service the call stack tree is 'clickable', was easier for me to debug and I just forgot to revert

File: projects/hslayers/src/components/save-map/layman.service.ts (hunk 5 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/save-map/layman.service.ts:1]: Once again a change I havent even realized I made. Must have missclicked 'Fix all prettier..' . Will get rid of that, thanks

File: projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts (hunk 4 above)
Comment (raitisbe) [reply projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts:117]: Will have to make hsLog also clickable.

Comment (jmacura) [reply projects/hslayers/src/components/add-data/file/shp/add-data-file-shp.service.ts:117]: I can see. I use the console.log for debugging as well. I only switch to hsLog when I feel that the log could be useful for debugging possible runtime errors on production. Yes, clickable hsLog would a nice thing to have 🙂

**Review by raitisbe — COMMENTED** [review]
Could you add BREAKING change conventional commit message to notify the users that new peer dependencies need to be added?

**Comment by FilipLeitner** [comment]
> Could you add BREAKING change conventional commit message to notify the users that new peer dependencies need to be added?

Yea sure, ill squash and add that

**Review by raitisbe — APPROVED** [review]

**Pull request merged**